src/foam/lib/formatter/JSONFObjectFormatter.java

```diff
@@ -338,15 +338,14 @@ public void outputDelta(FObject oldFObject, FObject newFObject, ClassInfo defaul
       append(',');
     }
     addInnerNewline();
-    PropertyInfo id = (PropertyInfo) newInfo.getAxiomByName("id");
-    boolean outputId = ! id.getStorageTransient();
-    if ( outputId ) outputProperty(newFObject, id);
+
+    // Always output id although it can be transient or multipart id to ensure
+    // that the entry/record is identifiable on replay.
+    outputId(newFObject, newInfo);
```

Comment:
I think that this method should be re-written to include the functionality of getDelta() in-line. getDelta creates an ArrayList, which will generate a lot of garbage. Also, it caches the value in delta_, which doesn't seems safe. Once we do that, we can modify the delta behaviour to just exclude id properties, which will actually make it faster still because we won't have to bother with the comparison.

Reply:
On update, getDelta() doesn't include the id property and the multipart-id's properties because id wasn't changed but  the output json would still need the identifier field(s) for replaying. The `outputId()` method ensures the id/multipart-id is written out.